lib/pbench/server/database/models/datasets.py

```diff
@@ -906,10 +918,15 @@ class Metadata(Database.Base):
     # --- Standard Metadata keys
 
     # Metadata keys that clients can update
-    USER_UPDATEABLE_METADATA = [DASHBOARD, USER]
+    USER_UPDATEABLE_METADATA = [DASHBOARD]
```

Comment:
"user" was never defined, and as we discussed at the time it's not even necessarily clear that it makes sense. I liked this for favorites, because mapping dataset-specific user context into the dataset queries makes a lot of sense.

We never addressed any real model for how "clients" might share a single key namespace, which means there really isn't much advantage in user and dashboard over just dashboard.

We can add a new "client" namespace that behaves like "dashboard" if we want, and there's a Jira to make that configurable in a server instance (e.g., via an admin API or config file)... which would let a server admin build a namespace for each client that wants one. Or we can just build a "client" and document some "good neighbor" sharing rules ... or maybe nobody will even care.

Reply:
I don't like the idea of needing a server admin to configure a per-client namespace.  I feel like, if a user can log into the server, s/he should be able to create whatever metadata his client needs (up to some size limit...).  If someone wants to bring their own client, I feel like that should be OK (but, as you say, we should outline some "good neighbor" rules).  I'm willing to support having a custom namespace for the Dashboard, because we like them best, but I think we should be making provisions for other clients (if for no other reason than to dissuade them from invading the `dashboard` namespace).

I think that once we have Nikhil's mechanism hooked up to the `user` namespace, it'll all make good sense.

(Maybe the mistake was naming that namespace "user" instead of "client", since user has other connotations in the domain.)

Reply:
Yeah, "user" wasn't the best name in the first place, and I like `user.favorite` in the new model regardless of contextual dispute about the other two. 😄 

I could easily create a "client" namespace now or later. We could easily allow dynamically creating namespaces with an API to register/list/delete dataset metadata namespaces. Right now I think that's probably overkill.

Reply:
I suppose we could create a new `user` namespace (with the understanding that the old one is changing its name to something to be determined when we actually implement it...).

Having an API to register/etc. new namespaces sounds cool, but I'm not sure it buys us anything over having clients just using their own node under `client`.

You said "dataset metadata namespaces"...I was thinking in terms of the values in the `client` namespace being tied to the logged in user and not to any specific dataset.  I'm not prepared to contemplate the possibilities of per-user/per-client metadata values being associated with individual datasets.  🤯  Is that where you are coming from?